src/box/alter.cc

```diff
@@ -1830,56 +1830,83 @@ alter_space_move_indexes(struct alter_space *alter, uint32_t begin,
 	return 0;
 }
 
+/**
+ * Helper of @sa update_view_references.
+ * Update view reference counter of a space by its given @a space_name,
+ * adding @a update_value to the counter.
+ * If @a suppress_error is true - always return 0; otherwise, and if the space
+ * was not found (or not applicable or etc) - diag is set and -1 is returned.
+ */
+static int
+update_view_space_reference(struct Select *select, const char *space_name,
+			    int update_value, bool suppress_error)
+{
+	if (space_name == NULL)
+		return 0;
+	/*
+	 * Views are allowed to contain CTEs. CTE is a temporary object,
+	 * created and destroyed at SQL runtime (it is represented by an
+	 * ephemeral table). So, it is absent in space cache and as a
+	 * consequence we can't increment its reference counter.
+	 * Skip iteration.
+	 */
+	if (sql_select_constains_cte(select, space_name))
+		return 0;
+	struct space *space = space_by_name0(space_name);
+	if (space == NULL) {
+		if (suppress_error)
+			return 0;
+		diag_set(ClientError, ER_NO_SUCH_SPACE, space_name);
+		return -1;
```

Comment:
According to the coveralls report, this never happens. Let's drop this code and inline this function.